packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -810,35 +819,63 @@ class ShortcutManager with Diagnosticable, ChangeNotifier {
 ///  * [Action], a class for defining an invocation of a user action.
 ///  * [CallbackAction], a class for creating an action from a callback.
 class Shortcuts extends StatefulWidget {
-  /// Creates a const [Shortcuts] widget.
+  /// Creates a const [Shortcuts] widget that owns the map of shortcuts and
+  /// creates its own manager.
+  ///
+  /// When using this constructor, [manager] will return null.
   ///
   /// The [child] and [shortcuts] arguments are required.
+  ///
+  /// See also:
+  ///
+  ///  * [Shortcuts.manager], a constructor that uses a [ShortcutManager] to
+  ///    manage the shortcuts list instead.
   const Shortcuts({
     super.key,
-    this.manager,
-    required this.shortcuts,
+    required Map<ShortcutActivator, Intent> shortcuts,
     required this.child,
     this.debugLabel,
-  }) : assert(shortcuts != null),
+  }) : _shortcuts = shortcuts,
+       manager = null,
+       assert(shortcuts != null),
+       assert(child != null);
+
+  /// Creates a const [Shortcuts] widget that uses the [manager] to
+  /// manage the map of shortcuts.
+  ///
+  /// If this constructor is used, [shortcuts] will return the contents of
+  /// [ShortcutManager.shortcuts].
+  ///
+  /// The [child] and [manager] arguments are required.
+  const Shortcuts.manager({
+    super.key,
+    required ShortcutManager this.manager,
+    required this.child,
+    this.debugLabel,
+  }) : _shortcuts = const <ShortcutActivator, Intent>{},
+       assert(manager != null),
        assert(child != null);
 
   /// The [ShortcutManager] that will manage the mapping between key
   /// combinations and [Action]s.
   ///
   /// If not specified, uses a default-constructed [ShortcutManager].
   ///
-  /// This manager will be given new [shortcuts] to manage whenever the
-  /// [shortcuts] change materially.
+  /// If this widget was created with [Shortcuts.manager], the
+  /// [ShortcutsManager.shortcuts] will be used as the source for shortcuts. If
+  /// the unnamed constructor is used, this manager will be given new
```

Comment:
this field will stay null if the unnamed constructor is used, no?

Reply:
Oh! yes. Duh.